src/plugins/workspace/public/components/workspace_form/utils.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DEFAULT_SELECTED_FEATURES_IDS } from '../../../common/constants';
+
+import { WorkspaceFeature, WorkspaceFeatureGroup, WorkspaceFormErrors } from './types';
+
+export const isWorkspaceFeatureGroup = (
+  featureOrGroup: WorkspaceFeature | WorkspaceFeatureGroup
+): featureOrGroup is WorkspaceFeatureGroup => 'features' in featureOrGroup;
+
+export const isDefaultSelectedFeatureId = (id: string) => {
+  return DEFAULT_SELECTED_FEATURES_IDS.indexOf(id) > -1;
+};
+
+export const appendDefaultFeatureIds = (ids: string[]) => {
+  // concat default checked ids and unique the result
+  return Array.from(new Set(ids.concat(DEFAULT_SELECTED_FEATURES_IDS)));
+};
+
+// Validate name and description fields according related field UI description
+export const isValidNameOrDescription = (input?: string) => {
```

Comment:
Maybe we can rename `isValidNameOrDescription` -> `isValidFormTextInput`, does it sound good?

Reply:
Can be renamed, seems all the text input are using the same rule for now.